### ISSUE TITLE
Disabled traits with fake-user are still active at start

### ISSUE THREAD

**Issue by MoritzBrueckner** [armory3d/armory#2327]
To keep track of what was mentioned along the way in https://github.com/armory3d/armory/issues/2289:

> Another issue in conjunction with this is if a trait is `exported` and not `enabled`, the trait is still active at start.
> ![image](https://user-images.githubusercontent.com/55564981/128166507-3ffc5163-26d6-4479-b6f1-7f1c4feb3bea.png)
> 
> This also means that disabling both `export` and `enable`, the trait is never exported and cannot be added at run time.
> ![image](https://user-images.githubusercontent.com/55564981/128166869-b0221ae8-927b-4cc5-a7e5-032f0e010d50.png)

**Comment by tong** [comment]
Isn't it the correct behaviour to **not** export the trait when not enabled and no fake user ?
Someone can still reference the trait from code or add `--macro keep('Some')` to force export.

**Comment by MoritzBrueckner** [comment]
> Isn't it the correct behaviour to not export the trait when not enabled and no fake user ?

Yes, maybe the description above is a bit misleading in that regard. Not the second case is the problem, but the first one: traits with fake user enabled were active at the start even though they were disabled, so the fake user basically just enabled the trait again. If a trait rotates an object and is disabled with fake user, the trait should be exported but the rotation shouldn't start immediately, but that's what happened before https://github.com/armory3d/armory/pull/2335.
